wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -94,7 +102,9 @@ getHistory cWalId accIds mAddrId = do
             | addr `S.member` accAddrs -> filterByAddrs (S.singleton addr)
             | otherwise                -> throw errorBadAddress
 
-    first filterFn <$> getFullWalletHistory cWalId
+    !res <- first filterFn <$> getFullWalletHistory cWalId
```

Comment:
This doesn't do what you think it does. This forces a pair but not its components! See comment above about where to force it.

Reply:
Oh dear, I haven't realised this code was using `first` from `Control.Arrow` and yielding a tuple. Of course, for a tuple forcing into WHNF won't do very much of a difference. Let me handle this.